apps/web/src/hooks/useBestAMMTrade.ts

```diff
@@ -103,10 +195,36 @@ export function useBestAMMTrade({ type = 'quoter', ...params }: useBestAMMTradeO
     autoRevalidate: quoterAutoRevalidate,
   })
 
-  return useMemo(
-    () => (isIceQuoterAPIEnabled? bestTradeFromIceQuoterApi : isQuoterAPIEnabled ? bestTradeFromQuoterApi : bestTradeFromQuoterWorker),
-    [bestTradeFromIceQuoterApi, bestTradeFromQuoterApi, bestTradeFromQuoterWorker, isIceQuoterAPIEnabled, isQuoterAPIEnabled],
-  )
+  useEffect(() => {
```

Comment:
rather than adding a global status, please add a field to the quote/trade indicating it's source

Reply:
Can you please explane more?